src/binary_operator.cpp

```diff
@@ -170,6 +170,113 @@ double eval_dbl(const binary_operator &bo, const std::unordered_map<std::string,
     }
 }
 
+void eval_batch_dbl(const binary_operator &bo, const std::unordered_map<std::string, std::vector<double>> &map,
+                    std::vector<double> &out_values)
+{
+    auto tmp = out_values;
+    eval_batch_dbl(bo.lhs(), map, out_values);
+    eval_batch_dbl(bo.rhs(), map, tmp);
+    switch (bo.op()) {
+        case binary_operator::type::add:
+            std::transform(out_values.begin(), out_values.end(), tmp.begin(), out_values.begin(), std::plus<double>());
```

Comment:
Missing ``#include <functional>``. Also, you can remove the type and just use ``std::plus<>()``.